superset/security/manager.py

```diff
@@ -1872,14 +1873,36 @@ def raise_for_access(
                         .one_or_none()
                     )
                     and dashboard_.roles
-                    and (slice_id := form_data.get("slice_id"))
                     and (
-                        slc := self.get_session.query(Slice)
-                        .filter(Slice.id == slice_id)
-                        .one_or_none()
+                        (
+                            # Native filter.
+                            form_data.get("type") == "NATIVE_FILTER"
+                            and (native_filter_id := form_data.get("native_filter_id"))
+                            and dashboard_.json_metadata
```

Comment:
hey @john-bodley I just tested this branch locally and I still see the permission error -- it seems that `dashboardId` is still not included in the filter's payload data (despite https://github.com/apache/superset/pull/25025). Curious to know if you are facing a different behavior on your end.

Reply:
@Vitor-Avila just to confirm have you rebuilt the frontend assets? If the problem still persists would you mind sharing a URL—leveraging the example dataset—so I can further debug this issue?

Reply:
Phew! I'm glad to hear that this was a false alarm.